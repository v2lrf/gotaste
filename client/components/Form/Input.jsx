import React, { Fragment } from 'react'
import PropTypes from 'prop-types'

function Input({
  label,
  type,
  name,
  value,
  autoComplete,
  inputRef,
  placeholder,
  onChange
}) {
  return (
    <Fragment>
      {/* eslint-disable-next-line */}
      <label className="block text-grey-darker mb-6 text-sm uppercase">
        {label}
        <input
          type={type}
          name={name}
          value={value}
          autoComplete={autoComplete}
          className="block w-full rounded w-full mt-1 py-2 px-3 border focus:outline-none focus:shadow-outline"
          ref={inputRef}
          placeholder={placeholder}
          onChange={onChange}
        />
      </label>
    </Fragment>
  )
}

Input.propTypes = {
  label: PropTypes.string.isRequired,
  type: PropTypes.oneOf(['text', 'email', 'password', 'tel', 'url']).isRequired,
  name: PropTypes.string.isRequired,
  value: PropTypes.string.isRequired,
  autoComplete: PropTypes.string,
  inputRef: PropTypes.func,
  placeholder: PropTypes.string,
  onChange: PropTypes.func.isRequired
}

Input.defaultProps = {
  autoComplete: '',
  inputRef: () => null,
  placeholder: ''
}

export default Input
