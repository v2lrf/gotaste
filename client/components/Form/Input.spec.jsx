import React from 'react'
import Input from './Input'

describe('Input', () => {
  let component
  let props
  beforeEach(() => {
    props = {
      label: 'Email',
      type: 'email',
      name: 'email',
      value: 'I am an initial value',
      onChange: jest.fn()
    }
    component = shallow(<Input {...props}>Some content!</Input>)
  })

  describe('Snapshots', () => {
    it('should match the default snapshot', () => {
      expect(component).toMatchSnapshot()
    })

    it('should match the snapshot with custom autoComplete', () => {
      component.setProps({ autoComplete: 'current-password' })
      expect(component).toMatchSnapshot()
    })

    it('should match the snapshot with a placeholder', () => {
      component.setProps({ placeholder: 'Enter some value in me' })
      expect(component).toMatchSnapshot()
    })
  })

  describe('Actions', () => {
    it('should call the onChange handler when a value changes', () => {
      component
        .find('input')
        .simulate('change', { target: { value: 'New text!' } })
      expect(props.onChange).toHaveBeenCalled()
    })
  })
})
