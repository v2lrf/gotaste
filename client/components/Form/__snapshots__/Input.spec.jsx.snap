// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input Snapshots should match the default snapshot 1`] = `
<Fragment>
  <label
    className="block text-grey-darker mb-6 text-sm uppercase"
  >
    Email
    <input
      autoComplete=""
      className="block w-full rounded w-full mt-1 py-2 px-3 border focus:outline-none focus:shadow-outline"
      name="email"
      onChange={[MockFunction]}
      placeholder=""
      type="email"
      value="I am an initial value"
    />
  </label>
</Fragment>
`;

exports[`Input Snapshots should match the snapshot with a placeholder 1`] = `
<Fragment>
  <label
    className="block text-grey-darker mb-6 text-sm uppercase"
  >
    Email
    <input
      autoComplete=""
      className="block w-full rounded w-full mt-1 py-2 px-3 border focus:outline-none focus:shadow-outline"
      name="email"
      onChange={[MockFunction]}
      placeholder="Enter some value in me"
      type="email"
      value="I am an initial value"
    />
  </label>
</Fragment>
`;

exports[`Input Snapshots should match the snapshot with custom autoComplete 1`] = `
<Fragment>
  <label
    className="block text-grey-darker mb-6 text-sm uppercase"
  >
    Email
    <input
      autoComplete="current-password"
      className="block w-full rounded w-full mt-1 py-2 px-3 border focus:outline-none focus:shadow-outline"
      name="email"
      onChange={[MockFunction]}
      placeholder=""
      type="email"
      value="I am an initial value"
    />
  </label>
</Fragment>
`;
